lib/internal/Magento/Framework/Api/Code/Generator/ExtensionAttributesGenerator.php

```diff
@@ -41,6 +47,7 @@ class ExtensionAttributesGenerator extends \Magento\Framework\Code\Generator\Ent
      */
     public function __construct(
         \Magento\Framework\Api\ExtensionAttribute\Config $config,
+        \Magento\Framework\Reflection\TypeProcessor $typeProcessor,
```

Comment:
In order to preserve constructor signature and do not introduce backward incompatible changes in Magento 2 we are using this approach:

`class ExistingClass
  
{
    /** @var \New\Dependency\Class */
    private $newDependency;
    
    /**
     * The getter function to get the new dependency for real application code
     *
     * @return \New\Dependency\Class
     *
     * @deprecated
     */
    private function getNewDependency()
    {
        if ($this->newDependency === null) {
            $this->newDependency = \Magento\Framework\App\ObjectManager::getInstance()->get(\New\Dependency\Class::class);
        }
        return $this->newDependency;
    }
    
    public function existingFunction()
    {
        // Existing functionality
        ...
        ...
       
        // Use $this->getNewDependency() wherever the new dependency is needed
        ...
        ...
    }
}`